### DISCUSSION TITLE
PyAEDT inside AEDT

### DISCUSSION THREAD

**Discussion by HLuo90** [ansys/pyaedt#2921] [Q&A] [answered]
In the documentation, it states:

PyAEDT works both inside AEDT and as a standalone application. This Python library automatically detects whether it is running in an IronPython or CPython environment and initializes AEDT accordingly. PyAEDT also provides advanced error management. Usage examples follow.

Do you have an example of how to import and use pyaedt inside AEDT? 

I see a lot of examples of pyaedt launching AEDT like a subprocess, but I don't see how you would use PyAEDT inside AEDT, for example inside AEDT: Tools -> Run Script -> Pyaedt_script.py (?)

**Comment by rjwilliams84** [comment, marked as answer]
If you have HFSS open already you can run scripts from python using the already open hfss instance by using the new_desktop_session=False flag in the pyaedt.launch_desktop(...) command. You'll need to define the project name and design names to match what you have open already if your aim is to use python scripting to manipulate or grab data from an existing design.

**Comment by HLuo90** [comment]
Okay, so it detects a currently running HFSS instance and interfaces with it, but does this need to be done with IronPython? This also works with CPython correct? 
I just wanted to make sure that pyAEDT does not support running inside HFSS.

**Reply by rjwilliams84** [reply]
I tried it out on my workstation and I just ran a script from an external IDE, not through hfss.

**Reply by MaxJPRey** [reply]
That is correct @rjwilliams84 and @HLuo90 .
If you have an opened AEDT session, you can interface with it from an external command prompt or IDE for instance and not necessarily from the IronPython command windows provided by AEDT gui.

**Comment by fwlfHS** [comment]
If you open `Tools -> Open Command Window` you get the IronPython prompt if you type in `from pyaedt import Hfss` it will use the IronPython version. Same applies to scripts that you run from within AEDT via `Tools -> Run Script...`

**Reply by HLuo90** [reply]
thanks! that works
